lib/pushmeup/apns/core.rb

```diff
@@ -27,20 +29,26 @@ def self.start_persistence
   
   def self.stop_persistence
     @persistent = false
-    
+
     @ssl.close
     @sock.close
   end
   
-  def self.send_notification(device_token, message)
+  def self.send_notification(device_token, message, options = {})
+    _logger_route = options.has_key?("rails_log_route") ? options[:rails_log_route] : STDOUT
+
+    @logger = Logger.new(_logger_route)
     n = APNS::Notification.new(device_token, message)
+    logger.debug "[Pushmeup::IOS::Send] Sending the following raw request #{n.packaged_notification}"
     self.send_notifications([n])
   end
   
   def self.send_notifications(notifications)
     @mutex.synchronize do
       self.with_connection do
```

Comment:
can you please add a debug to check SSL connection nil or not?

Also some debug logs when trying to create a SSL connection?